### DISCUSSION TITLE
How to register users with email verification?

### DISCUSSION THREAD

**Discussion by gromyko21** [fastapi-users/fastapi-users#1097] [Q&A]
How to register users with email verification?

**Comment by frankie567** [comment]
After registration, you can require the user to be verified in order to login: https://fastapi-users.github.io/fastapi-users/10.1/configuration/routers/auth/#optional-user-verification

To verify the email of a user, you need to setup the verification routes:

* Setup: https://fastapi-users.github.io/fastapi-users/10.1/configuration/routers/verify/
* Usage: https://fastapi-users.github.io/fastapi-users/10.1/usage/routes/#verify-router
